src/lib/coordination/collection.hpp

```diff
@@ -142,9 +148,43 @@ T list_idem_collection(node_t& node, trace_t call_point, real_t const& distance,
 //! @brief Export list for list_idem_collection.
 template <typename T> using list_idem_collection_t = common::export_list<T,real_t>;
 
+
+
+//! @brief Collects distributed data with a arithmetic strategy.
+template <typename node_t, typename T, typename G, typename = common::if_signature<G, T(T,T)>>
+T list_arith_collection(node_t& node, trace_t call_point, real_t const& distance, T const& value, real_t radius, real_t speed, T const& null, real_t epsilon, G&& accumulate) {
+    internal::trace_call trace_caller(node.stack_trace, call_point);
+
+    real_t t = node.current_time();
+    field<real_t> Tu = nbr(node, 1, node.next_time() + epsilon);
+    field<real_t> Pu = nbr(node, 2, distance + speed * (node.next_time() - t));
+    field<real_t> maxDistNow = node.nbr_dist() + speed * node.nbr_lag();
+    field<real_t> Vwst = mux(isfinite(distance) and maxDistNow < radius, (distance - Pu) / (Tu - t), (real_t)(-INF));
+
+    return old(node, 6, value, [&](field<T> x){
```

Comment:
how is x used? looks like it is completely ignored

Reply:
yes, I had already noticed that: my question was "is it **useful** at all?"

Reply:
Yes, because in x we have the partial aggregate results for each node that are needed to calculate the final results of the collection.

Reply:
the lambda that you pass to `old` takes a `field<T>` and returns a `T`
this is wrong, please look at `old` in the paper describing FCPP and/or in the source code definition of `old`
it works just because there's an implicit conversion from `T` to `field<T>` but the resulting field is constant, i.e., it maps any device to the same `T` value (try to print the `x` field to verify it)

Reply:
Ok, I solved the problem coverting x from field<T> to T. Now it takes a T and returns a T.